pkg/storage/cloudimpl/filetable/file_table_read_writer.go

```diff
@@ -163,18 +163,16 @@ type FileToTableSystem struct {
 
 // FileTable which contains records for every uploaded file.
 const fileTableSchema = `CREATE TABLE %s (filename STRING PRIMARY KEY, 
+file_id UUID UNIQUE NOT NULL DEFAULT gen_random_uuid(),
 file_size INT NOT NULL, 
 username STRING NOT NULL, 
 upload_time TIMESTAMP DEFAULT now())`
 
 // PayloadTable contains the chunked payloads of each file.
-// The Payload table is interleaved in the File table to prevent repetition
-// of filename for every chunk at the KV level.
-const payloadTableSchema = `CREATE TABLE %s (filename STRING, 
+const payloadTableSchema = `CREATE TABLE %s (file_id UUID, 
```

Comment:
I tried to put a REFERENCES constraint on file_id, but for some reason attempting to insert into the payload table after creation is returning a `table is being added error`. This doesn't make sense to me because the creation and insertion are in two separate transactions. Still looking into this, but would appreciate comments if something looks off.

Reply:
Stepping through the adding state is needed here to wait for the referenced metadata table to come online, but usually the conn executor is supposed to wait to ack your commit until the schema change steps to public. I wonder if internal exec doesn't?

Could get around this by creating both tables in same txn I think

Reply:
sorry just saw this, both tables are created in the same db.Txn closure in `createFileAndPayloadTables`. The weird part is that the test is able to write to the metadata table (which is the first thing we do when we issue a write), but errors out when trying to write the chunks to the payload table with the "table being added" error.